src/ios/PushPlugin.m

```diff
@@ -607,4 +620,97 @@ -(void)stopBackgroundTask:(NSTimer*)timer
     }
 }
 
+
+- (void)pushRegistry:(PKPushRegistry *)registry didUpdatePushCredentials:(PKPushCredentials *)credentials forType:(NSString *)type
+{
+    if([credentials.token length] == 0) {
+        NSLog(@"VoIPPush Plugin register error - No devide token:");
+        return;
+    }
+
+    NSLog(@"VoIPPush Plugin register success");
+    const unsigned *tokenBytes = [credentials.token bytes];
+    NSString *sToken = [NSString stringWithFormat:@"%08x%08x%08x%08x%08x%08x%08x%08x",
+                        ntohl(tokenBytes[0]), ntohl(tokenBytes[1]), ntohl(tokenBytes[2]),
+                        ntohl(tokenBytes[3]), ntohl(tokenBytes[4]), ntohl(tokenBytes[5]),
+                        ntohl(tokenBytes[6]), ntohl(tokenBytes[7])];
+
+    NSLog(@"VoIPPush Plugin device token %@", sToken);
```

Comment:
You can get rid of all the code from here until the end of the method and call:

```
[self registerWithToken:sToken];
```

to avoid code duplication.

Reply:
Ok